src/debug/validatePreDebug.ts

```diff
@@ -99,6 +111,29 @@ function getMatchingWorkspace(debugConfig: vscode.DebugConfiguration): vscode.Wo
     throw new Error(localize('noDebug', 'Failed to find launch config matching name "{0}", request "{1}", and type "{2}".', debugConfig.name, debugConfig.request, debugConfig.type));
 }
 
+/**
+ * Ensure that that Python V2+ projects have an appropriate version of Functions tools installed.
+ */
+async function validateFunctionVersion(context: IActionContext, projectLanguage: string | undefined, projectLanguageModel: number | undefined, workspacePath: string): Promise<boolean> {
+    const validateTools = getWorkspaceSetting<boolean>('validateFuncCoreTools', workspacePath) !== false;
+
+    if (validateTools && isPythonV2Plus(projectLanguage, projectLanguageModel)) {
+        const version = await getLocalFuncCoreToolsVersion(context, workspacePath);
+
+        // TODO: Update range with official version.
+        const expectedVersionRange = '>=4.0.4515';
+
+        if (version && !semver.satisfies(version, expectedVersionRange)) {
+            const message: string = localize('invalidFunctionVersion', 'The version of installed Functions tools "{0}" is not sufficient for this project type ("{1}")?', version, expectedVersionRange);
```

Comment:
The message ends in a question mark but isn't actually a question?

(teehee)

Reply:
It was supposed to be rhetorical.  :-) Updated to a boring, old period.